core/interfaces/i_draggable.ts

```diff
@@ -4,11 +4,63 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-// Former goog.module ID: Blockly.IDraggable
+import {Coordinate} from '../utils/coordinate';
+import {IDragTarget} from './i_drag_target';
 
-import type {IDeletable} from './i_deletable.js';
+/**
+ * Represents an object that can be dragged.
+ */
+export interface IDraggable extends IDragStrategy {
+  /** Returns true iff the element is currently movable. */
+  isMovable(): boolean;
+
+  /** Returns the current location of the draggable in workspace coordinates. */
+  getLocation(): Coordinate;
+}
 
 /**
- * The interface for an object that can be dragged.
+ * Represents an object that can be dragged.
  */
-export interface IDraggable extends IDeletable {}
+export interface IDragStrategy {
```

Comment:
Curious why you separated this out into a separate interface. I don't think it makes sense to have `BlockSvg` and `RenderedWorkspaceComment` implement `IDragStrategy` since that's not what they are. In my mind are draggable things, they have drag strategies.

Reply:
The purpose is so that I can have `class BlockDragStrategy implements IDragStrategy` and have it tell me if I made any mistakes with my type signatures.

I agree that it "`IDraggable extends DragStrategy`" looks a bit funny, but other than omitting `IDragStrategy` entirely I don't see any way that `BlockSvg` and `RenderedWorkspaceComment` won't both satisfy it, regardless of whether they formally implement it or not.

Reply:
Ah I missed that `IDraggable` was extending `IDragStrategy` this is fine!

Reply:
Actually, there is a way to avoid that: the methods could be given different names in `IDragStrategy` than in `IDraggable`.  But that seems… gratuitous.